### ISSUE TITLE
The Test Result of semantickitty  using the pretrained model is lower than paper 

### ISSUE THREAD

**Issue by Xavier-wa** [dvlab-research/SphereFormer#70]
Nice Work! But I have some problem about the infer result on semantick test set.First,I write the test_tta to infer like validate_tta on the source:
```
def test_tta(test_loader,model, criterion,args,learning_label_inv):
    if main_process():
        logger.info('>>>>>>>>>>>>>>>> Start Evaluation >>>>>>>>>>>>>>>>')
    data_time = AverageMeter()
    
    torch.cuda.empty_cache()
    
    path = args.eval_path
    model.eval()
    end = time.time()
    for i, batch_data_list in enumerate(test_loader):
        data_time.update(time.time() - end)
        

        with torch.no_grad():
            output = 0.0
            
            for batch_data in batch_data_list:

                (coord, xyz, feat, target, offset, inds_reconstruct,file_path) = batch_data
                inds_reconstruct = inds_reconstruct.cuda(non_blocking=True)

                offset_ = offset.clone()
                offset_[1:] = offset_[1:] - offset_[:-1]
                batch = torch.cat([torch.tensor([ii]*o) for ii,o in enumerate(offset_)], 0).long()

                coord = torch.cat([batch.unsqueeze(-1), coord], -1)
                spatial_shape = np.clip((coord.max(0)[0][1:] + 1).numpy(), 128, None)
            
                coord, xyz, feat, target, offset = coord.cuda(non_blocking=True), xyz.cuda(non_blocking=True), feat.cuda(non_blocking=True), target.cuda(non_blocking=True), offset.cuda(non_blocking=True)
                batch = batch.cuda(non_blocking=True)

                sinput = spconv.SparseConvTensor(feat, coord.int(), spatial_shape, args.batch_size_val)

                assert batch.shape[0] == feat.shape[0]

                output_i = model(sinput, xyz, batch)
                output_i = F.softmax(output_i[inds_reconstruct, :], -1)
                output = output + output_i

            output = output / len(batch_data_list)
            #get the max arg
            output = torch.argmax(output,1)
            output += 1  
            output[output==args.ignore_label+1] -=  (args.ignore_label+1)
            #mapping inv
            for i in range(len(learning_label_inv)):
                output[output==i] = learning_label_inv[i] + 100 #+100 to avoid some label be mapped twice
            output -= 100 

            label = output.cpu().numpy().astype(np.uint32)

            #Generated file
            file_seq,file_name = file_path[0][-22:-20],file_path[0][-10:-4]+'.label'
            output_path = f'{path}/{file_seq}/predictions/{file_name}'
            directory = os.path.dirname(output_path)
            if not os.path.exists(directory):
                os.makedirs(directory)
            label.tofile(output_path)
            logger.info(f'>>>>>>>>>>>>>>>> Generated {output_path} >>>>>>>>>>>>>>>>')
```
**but the infer result on Codalab  have a big grap:**
![image](https://github.com/dvlab-research/SphereFormer/assets/74730592/9ed7eb1d-ad7a-4b9c-8507-9fae9c1be02b)
**the details:**
`mean IoU,mean accuracy,car,bicycle,motorcycle,truck,other-vehicle,person,bicyclist,motorcyclist,road,parking,sidewalk,other-ground,building,fence,vegetation,trunk,terrain,pole,traffic-sign
67.4,91.4,96.6,62.2,52.2,47.2,49.1,72.4,75.9,38.7,90.3,59.8,76.1,35.8,93.1,70.4,85.7,72.9,70.6,63.7,67.7`
Could you tell me the detail of test inference method you use,plz.Or may be are there some error in my code,Thank You.
**this is my conf:**
```
DATA:
  data_name: semantic_kitti
  data_root: /mnt/storage/dataset/semanticKitti/dataset/  ## Fill in the data path 
  label_mapping: util/semantic-kitti.yaml
  classes: 19
  fea_dim: 6
  voxel_size: [0.05, 0.05, 0.05]
  voxel_max: 120000 

TRAIN:
  # arch
  arch: unet_spherical_transformer
  input_c: 4
  m: 32
  block_reps: 2
  block_residual: True
  layers: [32, 64, 128, 256, 256]
  quant_size_scale: 24
  patch_size: 1 
  window_size: 6
  use_xyz: True
  sync_bn: True  # adopt sync_bn or not
  rel_query: True
  rel_key: True
  rel_value: True
  drop_path_rate: 0.3
  max_batch_points: 1000000
  class_weight: [ 3.1557,  8.7029,  7.8281,  6.1354,  6.3161,  7.9937,  8.9704,
                          10.1922,  1.6155,  4.2187,  1.9385,  5.5455,  2.0198,  2.6261,  1.3212,
                          5.1102,  2.5492,  5.8585,  7.3929]
  xyz_norm: False
  pc_range: [[-51.2, -51.2, -4], [51.2, 51.2, 2.4]]
  window_size_sphere: [2, 2, 80]
  window_size_scale: [2.0, 1.5]
  sphere_layers: [1,2,3,4,5]
  grad_checkpoint_layers: []
  a: 0.0125
  loss_name: ce_loss
  use_tta: True
  vote_num: 4

  # training
  aug: True
  transformer_lr_scale: 0.1 
  scheduler_update: step 
  scheduler: Poly

  power: 0.9
  use_amp: True
  train_gpu: [0] 
  workers: 10  # data loader workers 
  batch_size: 12 # batch size for training
  batch_size_val:  1 # batch size for validation during training, memory and speed tradeoff
  base_lr: 0.006 
  epochs: 50 
  start_epoch: 0
  momentum: 0.9
  weight_decay: 0.02 
  drop_rate: 0.5

  ignore_label: 255
  manual_seed: 123
  print_freq: 10
  save_freq: 1
  save_path: runs/semantic_kitti_unet32_spherical_transformer
  weight:  /XXX/model_semantic_kitti.pth # path to initial weight (default: none)
  resume:  # path to latest checkpoint (default: none)
  evaluate: True  # evaluate on validation set, extra gpu memory needed and small batch_size_val is recommend
  eval_freq: 1
  val: True
  evalType: test
  eval_path: /XXX/sphereFormer/pretrained/sequences/
Distributed:
  dist_url: tcp://127.0.0.1:6789
  dist_backend: 'nccl'
  multiprocessing_distributed: True
  world_size: 1
  rank: 0
```

another I also want to know the infer speed when you use your code,I infer using my code on v100(32GB),But the speed is very low(about 1 label/s).

**Comment by Xavier-wa** [comment]
I guess maybe the pretrained model be release without gt-paste and  multi-model ensembling,but I am not sure

**Comment by 2003416** [comment]
您好，不知道您是否还保存着预训练权重，作者提供的预训练权重网页已经失效了。
